### PR TITLE
Refactor backward compatibility tests to use override_qengines decorator

### PR DESCRIPTION
**Pull request by durumu**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38838 Refactor backward compatibility tests to use override_qengines decorator**

Differential Revision: [D21676032](https://our.internmc.facebook.com/intern/diff/D21676032)

### CODE REVIEW

File: test/quantization/test_backward_compatibility.py [hunk 1]
```diff
@@ -11,13 +11,14 @@
 
 # Testing utils
 from torch.testing._internal.common_utils import TestCase
-from torch.testing._internal.common_quantized import override_quantized_engine, supported_qengines
+from torch.testing._internal.common_quantized import override_quantized_engine, supported_qengines, override_qengines
+
```
Comment (supriyar) [inline test/quantization/test_backward_compatibility.py:15]: Please fix the lint errors before submitting

File: test/quantization/test_backward_compatibility.py [hunk 2]
```diff
@@ -68,55 +69,50 @@ def remove_prefix(text, prefix):
...
-        for i, qengine in enumerate(supported_qengines):
-            with override_quantized_engine(qengine):
-                module_qint8 = nnqd.Linear(3, 1, bias_=True, dtype=torch.qint8)
-                self._test_op(module_qint8, "qint8", input_size=[1, 3], input_quantized=False, generate=False, iter=i)
-                if qengine == 'fbgemm':
-                    module_float16 = nnqd.Linear(3, 1, bias_=True, dtype=torch.float16)
-                    self._test_op(module_float16, "float16", input_size=[1, 3], input_quantized=False, generate=False)
+        module_qint8 = nnqd.Linear(3, 1, bias_=True, dtype=torch.qint8)
+        self._test_op(module_qint8, "qint8", input_size=[1, 3], input_quantized=False, generate=False)
+        if torch.backends.quantized.engine == 'fbgemm':
```
Comment (jerryzh168) [inline test/quantization/test_backward_compatibility.py:86]: we can have a function for this check as well

Comment (jerryzh168) [reply test/quantization/test_backward_compatibility.py:86]: please put it in torch/testing/_internal/common_quantization.py

File: test/quantization/test_backward_compatibility.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/test_backward_compatibility.py:15]: I think we should also move everything in common_quantized to common_quantization and remove that file. But this change should be a separate PR.

**Review by jerryzh168 — APPROVED** [review]
Looks good, thanks!

**Pull request closed without merging**